examples/adhoc/README.md

```diff
@@ -1,55 +1,52 @@
 ## Adhoc mode examples
 
-These examples showcase the different running modes of `pyroscope adhoc`.
-Check [adhoc documentation](https://pyroscope.io/docs/agent-configuration-adhoc) for complete information about what modes are supported by each language.
+Pyroscope provides three options for using the "adhoc" mode, depending on whether you have the Pyroscope SDK installed or not. Choose the appropriate method based on your setup.
 
-Note: The example programs are toy examples but they share some interesting properties: their running time should take around 1 minute, and their profiling data should change slightly from run to run (making them good examples for comparison / diff mode visualizations).
+### Option 1: Push mode (with SDK installed)
+If your application already uses an agent or has integration with the Pyroscope HTTP API, you can use push mode
+to profile the application without any additional configuration changes.
 
-### Push Mode
+#### Golang Adhoc (Push Mode)
 If the application to profile is already using an agent or has some integration with the HTTP API,
 push mode can be used to profile the application without any configuration change.
 
-#### Golang adhoc (push)
 ```
 # no spy is supported, --push is not needed (but can still be provided):
 pyroscope adhoc go run adhoc-push.go
 ```
+
 #### Python adhoc (using pip package)
 ```
 # pyspy is autodetected, --push is mandatory.
-# Note that you need pyroscope-io >= 0.6.0 for this to work.
 pyroscope adhoc --push python adhoc-push.py
 ```
 
+### Option 2: Push Mode (no SDK installed -- Pyroscope sidecar)
+
+If you don't have the Pyroscope SDK installed and want to profile a Python or Ruby application, you can still push data
+to the Pyroscope server using the adhoc mode as a sidecar.
 
-### Exec mode
+#### Python adhoc (no pip package -- Pyroscope sidecar)
 For languages with a supported spy and no other pyroscope integration, this is the easiest way to get profiling data.
 For example, this method will work for python or ruby _without_ the pip/gem instrumented in the code.
+For this you will need to use the `pyroscope-cli` from the [pyroscope-rs](https://github.com/grafana/pyroscope-rs/releases/tag/cli-0.2.7) project.
 
-```
-# Run with spy-name autodetected.
-pyroscope adhoc python adhoc-spy.py
+**Note: this is different than using the pyroscope binary, `./pyroscope`, itself.**
 
-# Alternatively, specify the spy-name if it cannot be autodetected.
-pyroscope adhoc --spy-name pyspy ./adhoc-spy.py
+Profile a script using adhoc
+```
+pyroscope adhoc --push pyroscope-cli exec --spy-name=pyspy main.py
 ```
 
-### Connect mode
-#### Profile a process that is already running
-
-If the profiled process is already running, it's possible to attach to it instead, indicating its PID through the `--pid` flag:
+Attach to a process and profile it using `pyroscope-cli`
+```
+sudo ./pyroscope-cli connect --pid=[369936] --spy-name=pyspy --server-address="http://localhost:4100"
```

Comment:
the above is not adhoc, but just sends data to pyroscope

This should work for adhoc
```suggestion
sudo pyroscope adhoc --push pyroscope-cli connect --spy-name=pyspy --pid=<pid>
```